### PR TITLE
Krita: 4.4.7 -> 4.4.8 + Add beta version 5.0.0-beta1

### PR DESCRIPTION
**Pull request by sky1e**
###### Motivation for this change
Update Krita package to latest stable patch version, as well as add a new variant of the package for the latest beta.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by artemist** [comment]
@GrahamcOfBorg build krita krita-beta
(I think I should be allowed to do this, let's see)

File: pkgs/applications/graphics/krita/default.nix [hunk 1]
```diff
@@ -6,15 +6,27 @@
...
+      sha256 = "1p5l2vpsgcp4wajgn5rgjcyb8l5ickm1nkmfx8zzr4rnwjnyxdbm";
+    } else {
+      version = "4.4.8";
+      stable = "stable";
+      sha256 = "1y0d8gnxfdg5nfwk8dgx8fc2bwskvnys049napb1a9fr25bqmimw";
+    };
   pname = "krita";
-  version = "4.4.7";
+in with build; mkDerivation {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/krita/default.nix:25]:
```suggestion
in mkDerivation {
  inherit (build) pname version;
```

File: pkgs/applications/graphics/krita/default.nix [hunk 2]
```diff
@@ -6,15 +6,27 @@
...
 
   src = fetchurl {
-    url = "https://download.kde.org/stable/${pname}/${version}/${pname}-${version}.tar.gz";
-    sha256 = "sha256-I6fFxPRCcRU5dyFXZPvGvTb9MuGikrvTaGCXpp4LRRk=";
+    url = "https://download.kde.org/${stable}/${pname}/${version}/${pname}-${version}.tar.gz";
+    inherit sha256;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/krita/default.nix:29]:
```suggestion
    inherit (build) sha256;
```

File: pkgs/applications/graphics/krita/default.nix [hunk 3]
```diff
@@ -6,15 +6,27 @@
...
+  build =
+    if beta then {
+      version = "5.0.0-beta1";
+      stable = "unstable";
+      sha256 = "1p5l2vpsgcp4wajgn5rgjcyb8l5ickm1nkmfx8zzr4rnwjnyxdbm";
+    } else {
+      version = "4.4.8";
+      stable = "stable";
+      sha256 = "1y0d8gnxfdg5nfwk8dgx8fc2bwskvnys049napb1a9fr25bqmimw";
+    };
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/krita/default.nix:22]: Why not use the common approach like php does?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by artemist — APPROVED** [review]
Everything looks resolved, LGTM

**Pull request merged**